### PR TITLE
patch: add istio-injection labels

### PR DESCRIPTION
**Pull request by bryanpaget**
Closes issue [#1093](https://github.com/StatCan/daaas/issues/1093).

I followed the approach taken by @rohank07 on the following commit:

https://github.com/StatCan/aaw-kubeflow-manifests/commit/ecf8a46bc0c732851df195ef6d868126b8a26325

### CODE REVIEW

**Review by cboin1996 — APPROVED** [review]
Looks good to me. Out of curiosity, did you verify the labels appear when applying the kustomize?

**Comment by bryanpaget** [comment]
> Looks good to me. Out of curiosity, did you verify the labels appear when applying the kustomize?

I ran `kustomize build .`, this is how the labels appear:

```
...
apiVersion: v1
data:
  namespace-labels.yaml: |
    istio-injection: "enabled"
    katib.kubeflow.org/metrics-collector-injection: "enabled"
    serving.kubeflow.org/inferenceservice: "enabled"
    pipelines.kubeflow.org/enabled: "true"
    app.kubernetes.io/part-of: "kubeflow-profile"
kind: ConfigMap
metadata:
  name: namespace-labels-data-dh44gbf9c5
  namespace: kubeflow
...
```

File: kustomize/apps/profiles/base/namespace-labels.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+katib.kubeflow.org/metrics-collector-injection: "enabled"
```
Comment (chuckbelisle) [inline kustomize/apps/profiles/base/namespace-labels.yaml:1]: need to add the istio label here `istio-injection: "enabled"`

**Review by chuckbelisle — COMMENTED** [review]
see inline comment, missing label in yaml file

File: kustomize/apps/profiles/base/namespace-labels.yaml (hunk 1 above)
Comment (cboin1996) [reply kustomize/apps/profiles/base/namespace-labels.yaml:1]: @bryanpaget can you open a quick PR adding `istio-injection: "enabled"`

Comment (bryanpaget) [reply kustomize/apps/profiles/base/namespace-labels.yaml:1]: It seems to be merged already on dev:

https://github.com/StatCan/aaw-kubeflow-manifests/blob/aaw-dev-cc-00/kustomize/apps/profiles/base/namespace-labels.yaml

Comment (bryanpaget) [reply kustomize/apps/profiles/base/namespace-labels.yaml:1]: But I am missing one of the labels. For instance, my PR for prod has the following labels:

```
istio-injection: "enabled"
katib.kubeflow.org/metrics-collector-injection: "enabled"
serving.kubeflow.org/inferenceservice: "enabled"
pipelines.kubeflow.org/enabled: "true"
app.kubernetes.io/part-of: "kubeflow-profile"
```

And dev has the following labels right now:

```
istio-injection: "enabled"
katib.kubeflow.org/metrics-collector-injection: "enabled"
serving.kubeflow.org/inferenceservice: "enabled"
pipelines.kubeflow.org/enabled: "true"
```

Should I add `app.kubernetes.io/part-of: "kubeflow-profile"` to dev?

**Pull request merged**